File: 792-binary-search/binary-search.java
class Solution {
    public int search(int[] arr, int k) {
        int n = arr.length;

        int left = 0, right = n - 1;
        
        while (left <= right) {
            int mid = (left + right) / 2;
            
            if (arr[mid] == k) {
                return mid;
            }
            
            if (arr[mid] > k) {
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }
        
        return -1;
        
    }
}

File: 792-binary-search/binary-search.cpp
class Solution {
public:
    int search(vector<int>& arr, int target) {
        int n = arr.size();

        int left = -1 , right = n - 1;
        while ( left + 1 < right ) {
            int mid = (left + right) / 2;
            if (arr[mid] < target) {
                left = mid;
            } else {
                right = mid;
            }
        }
        if (arr[right] == target)
            return right;
        else return -1;
    }
};